services/api/src/api/routes/webhook.py

```diff
@@ -125,6 +128,15 @@ async def webhook_endpoint(request: Request) -> Response:
                     logging.warning(f"/webhook: unexpected error while parsing the JSON body is invalid. Error: {e}")
                     return get_response(content, 500)
 
+                HEADER = "x-webhook-secret"
+                trust_sender = (
+                    hf_webhook_secret is not None
+                    and (secret := request.headers.get(HEADER)) is not None
+                    and secret == hf_webhook_secret
+                )
+                if not trust_sender:
```

Comment:
Maybe log `trust_sender` value anyway in before https://github.com/huggingface/datasets-server/pull/955/files#diff-75de3bd8cb5b5606e98fbf00f4e95bb2108ec49aab5fd00f7c97fb96734c7c90R66 for troubleshooting purposes.

Reply:
OK. As discussed directly, let's keep it as is for now. The idea is to log the anomalous behaviors, to avoid longing too much.